airflow/assets/__init__.py

```diff
@@ -276,20 +279,43 @@ class Asset(os.PathLike, BaseAsset):
     uri: str
     group: str
     extra: dict[str, Any]
+    watchers: list[BaseTrigger]
 
     asset_type: ClassVar[str] = "asset"
     __version__: ClassVar[int] = 1
 
     @overload
-    def __init__(self, name: str, uri: str, *, group: str = "", extra: dict | None = None) -> None:
+    def __init__(
+        self,
+        name: str,
+        uri: str,
+        *,
+        group: str = "",
+        extra: dict | None = None,
+        watchers: list[BaseTrigger] | None = None,
```

Comment:
I wonder if *watcher* is a good name for this. What do we expect this to do? If I understand AIP-82 correctly, an external event would fire the trigger, and the trigger would create events for assets associated to it.

Assuming my understanding is correct, the triggers here are not watchers of the asset; rather, the asset *watches* the triggers. The relationship is the other way around. So it is probably better to call this `watch` instead? Or maybe this attribute should live on the trigger instead, something like

```python
asset = Asset("example_asset_watchers")

trigger = SqsSensorTrigger(sqs_queue="my_queue", trigger=[asset])

DAG(..., schedule=[asset])
```

Tell me what you think on this.

Reply:
Naming ... so hard haha. I see your point.

The reason why I called it `watchers` is because the triggers will `watch` some external resource and send event on updates. In that sense, to me, the triggers are watchers. I am not strongly again `watch` if you think it makes more sense. To be very honest, between `watchers` and `watch` I dont mind, I think the both of them makes sense.

However, I definitely want the attribute on the asset class, I think it makes more sense and a more deliberate choice for the  user to say, I have this asset and I want this asset to be updated when these triggers fire.

Reply:
What do you think @uranusjr ?

Reply:
I did not hear back so I'll go with this. If any concerns or feedbacks, I can always adjust later